### PR TITLE
Add new controls closes #46

### LINKED ISSUE #46: Add new controls
**Is your feature request related to a problem? Please describe.**
Add below below controls:
     - kubernetes_cluster_os_disk_ephemeral
     - kubernetes_cluste_critical_pods_on_system_nodes
     - app_service_environment_zone_redundant
     - app_service_plan_zone_redundant
     - app_service_public_access_disabled
     - event_hub_namespace_latest_tls
     -  event_hub_namespace_zone_redundant
     - function_app_public_access_disabled
     - redis_cache_standard_replication_enabled
     - sql_database_ledger_enabled
     -sql_database_zone_redundant

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

**Review by rajlearner17 — APPROVED** [review]
LGTM

File: controls/appservice.sp [hunk 1]
```diff
@@ -477,3 +481,35 @@ control "appservice_function_app_builtin_logging_enabled" {
 
   tags = local.appservice_compliance_common_tags
 }
+
+control "appservice_plan_zone_redundant" {
+  title       = "App Service plans should be zone redundant"
+  description = "This control ensures that App Service plans is zone redundant."
```
Comment (madhushreeray30) [inline controls/appservice.sp]:
```suggestion
  description = "This control ensures that App Service plans are zone redundant."
```

File: controls/appservice.sp [hunk 2]
```diff
@@ -477,3 +481,35 @@ control "appservice_function_app_builtin_logging_enabled" {
...
+  query       = query.appservice_plan_zone_redundant
+
+  tags = local.appservice_compliance_common_tags
+}
+
+control "appservice_web_app_public_access_disabled" {
+  title       = "Web apps should restrict public network access"
+  description = "This control checks whether Web apps is not publicly accessible."
```
Comment (madhushreeray30) [inline controls/appservice.sp]:
```suggestion
  description = "This control checks whether Web apps are not publicly accessible."
```

File: controls/appservice.sp [hunk 3]
```diff
@@ -477,3 +481,35 @@ control "appservice_function_app_builtin_logging_enabled" {
...
+  title       = "App Service environment should be zone redundant"
+  description = "This control ensures that App Service environment is zone redundant."
+  query       = query.appservice_environment_zone_redundant_enabled
+
+  tags = local.appservice_compliance_common_tags
+}
+
+control "appservice_function_app_public_access_disabled" {
+  title       = "Function apps should restrict public network access"
+  description = "This control checks whether Function apps is not publicly accessible."
```
Comment (madhushreeray30) [inline controls/appservice.sp]:
```suggestion
  description = "This control checks whether Function apps are not publicly accessible."
```

File: controls/eventhub.sp [hunk 4]
```diff
@@ -9,8 +9,10 @@ benchmark "eventhub" {
   description = "This benchmark provides a set of controls that detect Terraform Azure Event Hub resources deviating from security best practices."
 
   children = [
+    control.event_hub_namespace_zone_redundant,
```
Comment (madhushreeray30) [inline controls/eventhub.sp]: Following the other control namings it should be `eventhub_namespace_zone_redundant`

File: controls/eventhub.sp [hunk 5]
```diff
@@ -36,3 +38,18 @@ control "eventhub_namespace_cmk_encryption_enabled" {
...
 
+control "eventhub_namespace_uses_latest_tls_version" {
+  title       = "Event Hub namespaces 'Minimum TLS version' should be set to 'Version 1.2'"
+  description = "This control checks whether 'Minimum TLS version' is set to 1.2. TLS 1.0 is a legacy version and has known vulnerabilities. This minimum TLS version can be configured to later protocols such as TLS 1.2."
+  query       = query.eventhub_namespace_uses_latest_tls_version
+
+  tags = local.eventhub_compliance_common_tags
+}
+
+control "event_hub_namespace_zone_redundant" {
```
Comment (madhushreeray30) [inline controls/eventhub.sp]:
```suggestion
control "eventhub_namespace_zone_redundant" {
```

File: controls/eventhub.sp [hunk 6]
```diff
@@ -36,3 +38,18 @@ control "eventhub_namespace_cmk_encryption_enabled" {
...
+  title       = "Event Hub namespaces should be zone redundant"
+  description = "This control ensures that Event Hub namespace is zone redundant."
+  query       = query.event_hub_namespace_zone_redundant
```
Comment (madhushreeray30) [inline controls/eventhub.sp]:
```suggestion
  query       = query.eventhub_namespace_zone_redundant
```

File: query/eventhub.sp [hunk 7]
```diff
@@ -57,3 +57,47 @@ query "eventhub_namespace_cmk_encryption_enabled" {
...
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      terraform_resource
+    where
+      type = 'azurerm_eventhub_namespace';
+  EOQ
+}
+
+query "event_hub_namespace_zone_redundant" {
```
Comment (madhushreeray30) [inline query/eventhub.sp]:
```suggestion
query "eventhub_namespace_zone_redundant" {
```

File: controls/kubernetes.sp [hunk 8]
```diff
@@ -188,3 +190,19 @@ control "kubernetes_cluster_node_pool_type_scale_set" {
 
   tags = local.kubernetes_compliance_common_tags
 }
+
+control "kubernetes_cluster_os_disk_ephemeral" {
+  title       = "Kubernetes clusters should use type ephemeral OS disk"
+  description = "Ensure Kubernetes clusters uses ephemeral type OS disk. This control is non-compliant if Kubernetes clusters do not use ephemeral type OS disk."
```
Comment (madhushreeray30) [inline controls/kubernetes.sp]:
```suggestion
  description = "Ensure Kubernetes clusters use ephemeral type OS disk. This control is non-compliant if Kubernetes clusters do not use ephemeral type OS disk."
```

File: controls/sql.sp [hunk 9]
```diff
@@ -23,7 +25,7 @@ benchmark "sql" {
     control.sql_server_azure_defender_enabled,
     control.sql_server_email_security_alert_enabled,
     control.sql_server_uses_latest_tls_version,
-    control.sql_server_vm_azure_defender_enabled
+    control.sql_server_vm_azure_defender_enabled,
```
Comment (madhushreeray30) [inline controls/sql.sp]:
```suggestion
    control.sql_server_vm_azure_defender_enabled
```

File: controls/sql.sp [hunk 10]
```diff
@@ -180,3 +182,19 @@ control "sql_database_log_monitoring_enabled" {
...
+control "sql_database_zone_redundant_enabled" {
+  title       = "SQL databases should be zone redundant"
+  description = "This control ensures that SQL database is zone redundant."
+  query       = query.sql_database_zone_redundant_enabled
+
+  tags = local.sql_compliance_common_tags
+}
+
+control "sql_database_ledger_enabled" {
+  title       = "SQL databases should ledger should be enabled"
```
Comment (madhushreeray30) [inline controls/sql.sp]:
```suggestion
  title       = "SQL database ledger should be enabled"
```

File: query/appservice.sp [hunk 11]
```diff
@@ -951,3 +951,91 @@ query "appservice_function_app_builtin_logging_enabled" {
...
+    select
+      address as resource,
+      case
+        when (attributes_std -> 'zone_balancing_enabled') is null then 'alarm'
+        when (attributes_std ->> 'zone_balancing_enabled')::bool then 'ok'
+        else 'alarm'
+      end status,
+      split_part(address, '.', 2) || case
+        when (attributes_std -> 'zone_balancing_enabled') is null then ' ''zone_balancing_enabled'' is not set'
+        when (attributes_std ->> 'zone_balancing_enabled')::bool then ' zone redundant enabled'
```
Comment (madhushreeray30) [inline query/appservice.sp]:
```suggestion
        when (attributes_std ->> 'zone_balancing_enabled')::bool then ' zone redundant'
```

File: query/appservice.sp [hunk 12]
```diff
@@ -951,3 +951,91 @@ query "appservice_function_app_builtin_logging_enabled" {
...
+        else ' zone redundant disabled'
```
Comment (madhushreeray30) [inline query/appservice.sp]:
```suggestion
        else ' not zone redundant'
```

File: query/appservice.sp [hunk 13]
```diff
@@ -951,3 +951,91 @@ query "appservice_function_app_builtin_logging_enabled" {
...
+    select
+      address as resource,
+      case
+        when (attributes_std -> 'zone_redundant') is null then 'alarm'
+        when (attributes_std ->> 'zone_redundant')::bool then 'ok'
+        else 'alarm'
+      end status,
+      split_part(address, '.', 2) || case
+        when (attributes_std -> 'zone_redundant') is null then ' ''zone_redundant'' is not set'
+        when (attributes_std ->> 'zone_redundant')::bool then ' zone redundant enabled'
```
Comment (madhushreeray30) [inline query/appservice.sp]:
```suggestion
        when (attributes_std ->> 'zone_redundant')::bool then ' zone redundant'
```

File: query/appservice.sp [hunk 14]
```diff
@@ -951,3 +951,91 @@ query "appservice_function_app_builtin_logging_enabled" {
...
+        else ' zone redundant disabled'
```
Comment (madhushreeray30) [inline query/appservice.sp]:
```suggestion
        else ' not zone redundant'
```

File: query/eventhub.sp [hunk 15]
```diff
@@ -57,3 +57,47 @@ query "eventhub_namespace_cmk_encryption_enabled" {
...
+    select
+      address as resource,
+      case
+        when (attributes_std ->> 'minimum_tls_version') is null then 'ok'
+        when (attributes_std ->> 'minimum_tls_version')::text = '1.2' then 'ok'
+        else 'alarm'
+      end status,
+      split_part(address, '.', 2) || case
+        when (attributes_std ->> 'minimum_tls_version') is null then ' TLS version not defined by default uses 1.2'
+        when (attributes_std ->> 'minimum_tls_version')::text = '1.2' then ' TLS version 1.2'
```
Comment (madhushreeray30) [inline query/eventhub.sp]:
```suggestion
        when (attributes_std ->> 'minimum_tls_version')::text = '1.2' then ' use TLS version 1.2'
```

File: query/eventhub.sp [hunk 16]
```diff
@@ -57,3 +57,47 @@ query "eventhub_namespace_cmk_encryption_enabled" {
...
+        else ' TLS version not 1.2'
```
Comment (madhushreeray30) [inline query/eventhub.sp]:
```suggestion
        else ' does not use TLS version 1.2'
```

File: query/eventhub.sp [hunk 17]
```diff
@@ -57,3 +57,47 @@ query "eventhub_namespace_cmk_encryption_enabled" {
...
+query "event_hub_namespace_zone_redundant" {
+  sql = <<-EOQ
+    select
+      address as resource,
+      case
+        when (attributes_std ->> 'zone_redundant')::bool then 'ok'
+        else 'alarm'
+      end status,
+      split_part(address, '.', 2) || case
+        when (attributes_std ->> 'zone_redundant')::bool then ' zone redundant enabled'
```
Comment (madhushreeray30) [inline query/eventhub.sp]:
```suggestion
        when (attributes_std ->> 'zone_redundant')::bool then ' zone redundant'
```

File: query/eventhub.sp [hunk 18]
```diff
@@ -57,3 +57,47 @@ query "eventhub_namespace_cmk_encryption_enabled" {
...
+        else ' zone redundant disabled'
```
Comment (madhushreeray30) [inline query/eventhub.sp]:
```suggestion
        else ' not zone redundant'
```

File: query/kubernetes.sp [hunk 19]
```diff
@@ -344,3 +344,45 @@ query "kubernetes_cluster_node_pool_type_scale_set" {
...
+query "kubernetes_cluster_os_disk_ephemeral" {
+  sql = <<-EOQ
+    select
+      address as resource,
+      case
+        when (attributes_std -> 'default_node_pool' ->> 'os_disk_type') = 'Ephemeral' then 'ok'
+        else 'alarm'
+      end status,
+      split_part(address, '.', 2) || case
+        when (attributes_std -> 'default_node_pool' ->> 'os_disk_type') = 'Ephemeral' then ' using ephemeral OS disks'
```
Comment (madhushreeray30) [inline query/kubernetes.sp]:
```suggestion
        when (attributes_std -> 'default_node_pool' ->> 'os_disk_type') = 'Ephemeral' then ' use ephemeral OS disks'
```

File: query/kubernetes.sp [hunk 20]
```diff
@@ -344,3 +344,45 @@ query "kubernetes_cluster_node_pool_type_scale_set" {
...
+        else  ' not using ephemeral OS disks'
```
Comment (madhushreeray30) [inline query/kubernetes.sp]:
```suggestion
        else  ' does not use ephemeral OS disks'
```

**Review by madhushreeray30 — COMMENTED** [review]
@khushboo9024 please take a look at the comments, thanks!

**Review by madhushreeray30 — APPROVED** [review]
LGTM 👍

**Pull request merged**